pinot-common/src/main/java/org/apache/pinot/common/utils/config/TableConfigUtils.java

```diff
@@ -212,4 +215,54 @@ public static ZNRecord toZNRecord(TableConfig tableConfig)
     znRecord.setSimpleFields(simpleFields);
     return znRecord;
   }
+
+  /**
+   * Helper method to convert from indexingConfig (that is deprecated) to ingestionConfig.
+   * Since we are moving from indexingConfig to ingestionConfig for ingestion related configs,
+   * this utility helps with that migration. It also handles corner cases where both these
+   * configs might exist, in which case ingestionConfig takes precedence, and only required
+   * fields absent there are carried over from indexingConfig.
+   *
+   * The conversion happens in-place, the specified tableConfig is mutated in-place.
+   *
+   * @param tableConfig Input table config.
+   */
+  public static void convertFromIndexToIngestionConfig(TableConfig tableConfig) {
```

Comment:
Shall we rename it to `convertFromLegacyTableConfig`? We can keep updating this method when we need to deprecate certain configs